From: Luca Weiss <luca.weiss@fairphone.com>
Date: Fri, 10 Jun 2022 09:51:11 +0200
Subject: arm64: dts: qcom: sm7225-fairphone-fp4: Enable speaker

Currently using downstream amp driver bindings. Need to be replaced!

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
---
 arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts | 120 +++++++++++++++++++++-
 1 file changed, 118 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
index d334a2a..887fc11 100644
--- a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
+++ b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
@@ -857,8 +857,46 @@ &i2c0 {
 
 	/* ST21NFCD NFC @ 8 */
 	/* VL53L3 ToF @ 29 */
-	/* AW88264A amplifier @ 34 */
-	/* AW88264A amplifier @ 35 */
+
+	aw882xx_l: aw882xx_smartpa@34 {
+		compatible = "awinic,aw882xx_smartpa";
+		reg = <0x34>;
+		reset-gpio = <&tlmm 12 0>;
+		irq-gpio = <&tlmm 48 0x2008>;
+		monitor-flag = <1>;
+		sync-flag = <0>;
+		monitor-timer-val = <3000>;
+		aw-cali-mode = "aw_misc";
+		sound-channel = "left";
+		aw-stereo-cali;
+		pinctrl-names = "aw_reset_low",
+			"aw_reset_high",
+			"aw_irq_active";
+		pinctrl-0 = <&aw_reset1_low>;
+		pinctrl-1 = <&aw_reset1_high>;
+		pinctrl-2 = <&aw_int1_low>;
+		#sound-dai-cells = <0>;
+	};
+
+	aw882xx_r: aw882xx_smartpa@35 {
+		compatible = "awinic,aw882xx_smartpa";
+		reg = <0x35>;
+		reset-gpio = <&tlmm 47 0>;
+		irq-gpio = <&tlmm 11 0x2008>;
+		monitor-flag = <1>;
+		sync-flag = <0>;
+		monitor-timer-val = <3000>;
+		aw-cali-mode = "aw_misc";
+		sound-channel = "right";
+		aw-stereo-cali;
+		pinctrl-names = "aw_reset_low",
+			"aw_reset_high",
+			"aw_irq_active";
+		pinctrl-0 = <&aw_reset2_low>;
+		pinctrl-1 = <&aw_reset2_high>;
+		pinctrl-2 = <&aw_int2_low>;
+		#sound-dai-cells = <0>;
+	};
 };
 
 &i2c8 {
@@ -1385,10 +1423,22 @@ &sdhc_2 {
 	status = "okay";
 };
 
+&q6afedai {
+	dai@127 {
+		reg = <QUINARY_MI2S_RX>;
+		qcom,sd-lines = <0>;
+	};
+};
+
 &sound {
 	compatible = "fairphone,fp4-sndcard";
 	model = "Fairphone 4";
 
+	pinctrl-0 = <&i2s1_active>;
+	pinctrl-1 = <&i2s1_sleep>;
+	pinctrl-names = "default",
+			"sleep";
+
 	mm1-dai-link {
 		link-name = "MultiMedia1";
 
@@ -1412,6 +1462,22 @@ platform {
 			sound-dai = <&q6routing>;
 		};
 	};
+
+	i2s-dai-link {
+		link-name = "I2S Playback";
+
+		cpu {
+			sound-dai = <&q6afedai QUINARY_MI2S_RX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&aw882xx_l>, <&aw882xx_r>;
+		};
+	};
 };
 
 &tlmm {
@@ -1536,6 +1602,56 @@ pmx_ts_release: ts-release-state {
 		drive-strength = <2>;
 		bias-disable;
 	};
+
+	aw-reset-int1-state {
+		aw_reset1_low: reset-low-pins {
+			pins = "gpio12";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+			output-low;
+		};
+
+		aw_reset1_high: reset-high-pins {
+			pins = "gpio12";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+
+		aw_int1_low: int-low-pins {
+			pins = "gpio48";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
+	aw-reset-int2-state {
+		aw_reset2_low: reset-low-pins {
+			pins = "gpio47";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+			output-low;
+		};
+
+		aw_reset2_high: reset-high-pins {
+			pins = "gpio47";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+
+		aw_int2_low: int-low-pins {
+			pins = "gpio11";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
 };
 
 &uart1 {
